### ISSUE TITLE
Doxygen 1.10 and 1.11 not documenting C# enum and its members

### ISSUE THREAD

**Issue by sjasionowska** [doxygen/doxygen#11012]
**Describe the bug**

After updating doxygen to 1.10, it stopped generating C# enum definitions. 
I tried using doxygen 1.11, and the result is the same. 
With 1.9 everything worked fine.

I can reference the enum and its members, so they are visible to the generator, but the generated links provide to the enum's namespace only. 

**To Reproduce**

1. Make sure you're using doxygen 1.10 or 1.11. 
2. Copy the enum code presented below.

```
namespace MyNamespace
{
	/// <summary>
	/// Sample enum.
	/// </summary>
	public enum MyEnum
	{
		/// <summary>
		/// First.
		/// </summary>
		First,
		
		/// <summary>
		/// Second.
		/// </summary>
		Second,
		
		/// <summary>
		/// Third.
		/// </summary>
		Third
	}
}
```

3. Try to generate the documentation of an enum presented below.
4. Reference the enum or one of its members. You'll get a link to the `MyNamespace` only, not the enum or its members.

**Expected behavior**

The enum scripting reference should be generated and accessible.

**Version**

doxygen 1.10 & 1.11 - doesn't work
doxygen 1.9 - works correctly

I'm using macOS Sonoma 14.5 (23F79)

Please let me know if more info is needed.

**Comment by albert-github** [comment]
I ran the given code with a default doxygen setting file and with the 1.9.8 version (always give the full version) and the 1.11.0 version I see in both cases:
![image](https://github.com/user-attachments/assets/3f24da23-27de-47b5-b3ec-e474b98c41ff)

I don't see a problem with the links either.
What did I miss (please indicate in an image)?
Furthermore:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).


(The results look OK to me as well with the current master version (1.12.0 (0b557b30eee328189629b2d3cc89f4774f3ad5ed))

**Comment by sjasionowska** [comment]
Hi Albert, after further investigation, it seems like the issue is on our side. Thank you for your time.

**Comment by reklamasieciowa** [comment]
I found two potential solutions to the problem. Both work. I don't know why. :upside_down: 

1) Removing the [<namespace> declaration](https://github.com/doxygen/doxygen/blob/master/templates/general/layout_default.xml#L106-L137). So I thought that we have a bad declaration and after deletion Doxygen takes a default one, but:
- all elements of the declaration are correct,
- I copied the <namespace> declaration from defalutLayout.xml and it doesn't work either.


To sum up: If in layout.xml we have:
<navindex>
<tab type=“namespaces” visible=“**no**” title=“”>.
then namespaces are generated **only if there is no** <namespace> definition, which looks like an error.

2) Change[ namespace declaration in navigation](https://github.com/doxygen/doxygen/blob/master/templates/general/layout_default.xml#L13) to visible=“yes”. Then it generates correctly, according to the <namespace> declaration in layout.xml.

**Comment by albert-github** [comment]
@reklamasieciowa 

When I understand it correctly you change your layout file into an own layoutfile with (say): `LAYOUT_FILE = mylayout.xml` and change the `mylayout.xml`:
- toggle `<tab type="namespaces" visible="yes" title="">` and `<tab type="namespaces" visible="no" title="">`

I don't get what you mean with 
> Generate documentation with and without namespace page declaration.

- Do you mean: leave / remove the part `<namespace> ... </namespace>`

To be sure we are talking about the same:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

**Comment by reklamasieciowa** [comment]
[example.zip](https://github.com/user-attachments/files/16389319/example.zip)

I attach an example.

LAYOUT_FILE = Input/Doxygen/layout.xml
Enum view: example/Output/html/namespace_my_namespace.html

3 cases:
1) tab type=“namespaces” visible=“**no**”, Layout definition for a namespace page **is given**. Enums are not generated.
2) tab type=“namespaces” visible=“**no**”, Layout definition for a namespace page **is not given / is commented out**. Enums are generated.
3) tab type=“namespaces” visible=“**yes**”, Layout definition for a namespace page is not required. It may be there, it may not be there. Enums are generated.

**Comment by reklamasieciowa** [comment]
I agree, but the problem is that **defining Layout definition** for a namespace page results in **NOT displaying** these elements. :)

**Comment by albert-github** [comment]
Apparently I'm missing something, so I created 4, full, layout files with biggest part of the <`navindex>` set to `visible=no`.
- in the versions 2 and 4 the `<namespace>`
- in the versions 1 and 2 the `<tab type="namespaces" visible="bo" title="">` and in 3 and 4 `<tab type="namespaces" visible="yes" title="">` 

Which one are you referring to?

**Comment by reklamasieciowa** [comment]
1. tab type=“namespaces” visible=“no”, Layout definition for a namespace page is given. Enums are not generated. Only when we remove the namespace page definition, it generates normally.
[layout.zip](https://github.com/user-attachments/files/16392280/layout.zip)

**Comment by albert-github** [comment]
The layout.zip you showed in https://github.com/doxygen/doxygen/issues/11012#issuecomment-2252756489 just is a stripped version of a full layout file and "corresponds" to the full_1.xml 

- What happens when you use this full_1.xml ?
- I don't think that it is necessary to generate the Enums as they are part of the Namespace.
- How would you jump to the Enums?

**Comment by reklamasieciowa** [comment]
I generated the documentation using https://github.com/doxygen/doxygen/blob/master/templates/general/layout_default.xml. The only change is tab type=“namespaces” visible=“**no**” title=“”. **Same thing, no enums.**

**Comment by albert-github** [comment]
> I don't think that it is necessary to generate the Enums as they are part of the Namespace.

**Comment by reklamasieciowa** [comment]
We just need the list of enums. 
![image](https://github.com/user-attachments/assets/6e5bb70e-45d7-432f-837c-4d0d47e57f54)
Before upgrading to 1.10.0, it generated correctly even with tab type=“namespaces” visible=“no”.

**Comment by albert-github** [comment]
- from which version did you upgrade? (I tried a few older versions with the indicated layout file but was not successful)
- How would you jump to the Enums?

What I see in the different versions is (html/index.html):
![image](https://github.com/user-attachments/assets/9e0477ff-37c7-4773-8305-0b2acbd8d51a)

**Comment by reklamasieciowa** [comment]
Updated from 1.9.8 What do you mean by "jump to the Enums"?

**Comment by albert-github** [comment]
I mean
- what is the address of the page like I mentioned I showed the page "html/index.html" which is the startup page
- what clicks etc. did you do to get the page you mentioned

**Comment by reklamasieciowa** [comment]
I wrote this in a comment https://github.com/doxygen/doxygen/issues/11012#issuecomment-2252310110 :)
Enum view: example/Output/html/namespace_my_namespace.html

Here you can see the difference. 
Documentation built with Layout definition for a namespace page -> no enums, blank page.
Documentation built without Layout definition for a namespace page -> there are enums.

![image](https://github.com/user-attachments/assets/85e15fb4-3bf5-462a-bc43-d6e6f4ed6d0f)

**Comment by albert-github** [comment]
> I wrote this in a comment https://github.com/doxygen/doxygen/issues/11012#issuecomment-2252310110 :)
> Enum view: example/Output/html/namespace_my_namespace.html

Sorry forgot about that.

Unfortunately the black images are hardly readable and I cannot enlarge them either, but I see that the difference is the `<!--` / `-->` around the `<namespace>`.

I get the same results when looking at the the html/namespace_my_namespace.html pages of both instances (which in general are the cases full_1 and full2 that I have).

I still haven't found how you went to these pages:
- what clicks etc. did you do to get the page you mentioned

**Comment by reklamasieciowa** [comment]
Yes, the only change is to comment out namespace definition.
I removed all menu items from layout.xml, so in this version it is impossible to go there by clicking. 
Use this file and click scripting -> MyNamespace
[layout.zip](https://github.com/user-attachments/files/16394422/layout.zip)

**Comment by albert-github** [comment]
OK I see now a new element went into the equation and indeed in this way it is possible to get to the mentioned page.

As we saw there is a difference between the situation when the part `<namespace>` is present and when not (and when the `<namespace>` present there is no difference between the situation where the `visible` is missing or has the value `yes` or `no`)

@doxygen 
- what would be the correct behavior in this case?

**Comment by reklamasieciowa** [comment]
In the previous version, everything worked with tab type=“namespaces” visible=“**no**” (we don't want to have this item visible in the menu) and namespace declaration set.
We would like it to still work to generate enum documentation in a situation where type=“namespaces” visible=“**no**” and the Layout definition for a namespace page declaration is given in Doxygen 1.11. :)

**Comment by albert-github** [comment]
@doxygen 
I'm wondering whether the initialization as done in `void LayoutDocManager::init()`:
```
  constexpr auto layoutFile = "layout_default.xml";
  QCString layout_default = ResourceMgr::instance().getAsString(layoutFile);
  parser.parse(layoutFile,layout_default.data(),Debug::isFlagSet(Debug::Lex_xml),
               [&]() { DebugLex::print(Debug::Lex_xml,"Entering","libxml/xml.l",layoutFile); },
               [&]() { DebugLex::print(Debug::Lex_xml,"Finished", "libxml/xml.l",layoutFile); }
              );
```
should be done when an explicit `LAYOUT_FILE` is given or the implicit file `DoxygenLayout.xml` is present.

**Comment by doxygen** [comment]
@albert-github The idea with the default layout file, is the same as with the default config file. If the user specified file is missing elements, the defaults from the default layout file are used. I could not really distill the problem from the (long) discussion yet.

**Comment by reklamasieciowa** [comment]
The main problemi specified here: https://github.com/doxygen/doxygen/issues/11012#issuecomment-2253061271 on the screenshot.

Long story short:
1) If tab type=“namespace” has visible=“no”, the enum documentation is not generated if the layout.xml file contains any namespace page definition. We checked with our definition and the definition from the default file (https://github.com/doxygen/doxygen/blob/master/templates/general/layout_default.xml).
As we comment out/remove the namespace page definition the enums generate correctly.
2) If tab type=“namespace” has visible=“yes”, then namespace page definition is not required. Whether it is there or not, the enums are generated correctly.

**Comment by albert-github** [comment]
@doxygen 
> The idea with the default layout file, is the same as with the default config file. If the user specified file is missing elements, the defaults from the default layout file are used.

That was clear to me, though I see differences in the output of 2 cases that should give the same results hence the `-d layout` debug option I created.

I thought I had attached once my set of used tests (layot / full versions 1-4 but this doesn't look like the to be the case so: [example_2.tar.gz](https://github.com/user-attachments/files/16453822/example_2.tar.gz)

I think especially the differences created, in respect to the layout debug output and the content of the page namespace_my_namespace.html, reachable throut the search for e.g MyEnum) by layout_1 and layout_2 are of importance.
These 2 cases only differ in the fact that the `<namespace>` part is commented out.

**Comment by sjasionowska** [comment]
I agree with @reklamasieciowa , we have the same issue.

**Comment by doxygen** [comment]
@sjasionowska Not yet, but I'm looking into it.

**Comment by doxygen** [comment]
Looks like it is related to [this PR](https://github.com/doxygen/doxygen/pull/9290), in particular this part in `startTop()` which seems to couple the navigation tabs to the layout of the pages (which should be independent):

```c++
      auto *lne = LayoutDocManager::instance().rootNavEntry()->find(nav);
      if (lne)
      {
        m_visible = m_visible && lne->visible();
        lne->setVisible(m_visible);
        recurseNavEntryChildren(*lne);
      }
```


@albert-github Any idea why this was added like this?

**Comment by albert-github** [comment]
I don't remember the actual reason but it might have been the, possible, incorrect idea that when e.g. the `class` was set to yes but the tab was not directly reachable (as it was disabled by the layout file) so from treeview / blue bar that in that case also the `class` page is not reachable anymore. 

I'm not sure what will happen when the layout file is shortened by removing items from it and the defaults do kick in.

Most likely the test should not be present (resulting in files that might not be used, not sure about what happens e.g. in latex or the other output formats).

**Comment by doxygen** [comment]
@sjasionowska Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by albert-github** [comment]
@doxygen 
I had a bit of a "strange" case lying around from #9290 where in the layout file there was in the class part (around line 82) an extra:
```
    <memberdecl visible="yes">
      <publicmethods visible="yes" title=""/>
    </memberdecl>
```
Which resulted with the 1.12.0 version in:
![image](https://github.com/user-attachments/assets/4697a900-ba27-424e-9a26-1a5a71705d44)

and with the current master (1.13.0 (c27ceadfa801eb7faee91dda5d71ececefb49ada)) in:
![image](https://github.com/user-attachments/assets/93ea4f7f-ecee-4e0f-a58d-0e05abd6a53c)


so missing the second mentioning of the public methiods.

Example: [example.tar.gz](https://github.com/user-attachments/files/18218206/example.tar.gz)

**Comment by doxygen** [comment]
@albert-github I added the duplication check for an earlier attempt to fix the issue, but I see that with the new merge algorithm it is not really needed. Still a not very useful scenario of course.

**Comment by albert-github** [comment]
> Still a not very useful scenario of course.

Correct, but one never knows and when the user asks and it is possible ...

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
